packages/PowerGrid/src/fit/fit-curve.ts

```diff
@@ -0,0 +1,241 @@
+import * as DG from 'datagrok-api/dg';
+import * as grok from 'datagrok-api/grok';
+import {
+  fit,
+  sigmoid,
+  FitErrorModel,
+  FitResult
+} from '@datagrok-libraries/statistics/src/parameter-estimation/fit-curve';
+import {GridColumn} from 'datagrok-api/dg';
+import {getChartData} from './fit-data';
+
+
+export function scaleCoordinates(coordinates: {x: number[], y: number[]}, canvasWidth: number, canvasHeight: number):
+  {x: number[], y: number[]} {
+  const xMin = Math.min(...coordinates.x);
+  const xMax = Math.max(...coordinates.x);
+  const yMin = Math.min(...coordinates.y);
+  const yMax = Math.max(...coordinates.y);
+
+  const xDiff = xMax - xMin;
+  const yDiff = yMax - yMin;
+  const xMid = (xMax + xMin) / 2;
+  const yMid = (yMax + yMin) / 2;
+
+  const coeff = xDiff * canvasHeight >= yDiff * canvasWidth ? canvasWidth / xDiff : canvasHeight / yDiff;
+  const canvasXCenter = canvasWidth / 2;
+  const canvasYCenter = canvasHeight / 2;
+
+  const scaledCoordinates: {x: number[], y: number[]} = {x: [], y: []};
+  for (let i = 0; i < coordinates.x.length; i++) {
+    scaledCoordinates.x[i] = canvasXCenter + coeff * (coordinates.x[i] - xMid);
+    scaledCoordinates.y[i] = canvasYCenter + coeff * (coordinates.y[i] - yMid);
+  }
+
+  return scaledCoordinates;
+}
```

Comment:
about this function scaleCoordinates
not sure this is right approach
probably best would be if we stored only native data and all other stuff like mapping to screen would be done separately by transformation (methods xToScreen, yToScreen, screenToX, screenToY, hitTest)
Check out how it is done for scatterplot and other viewers
Thoug for now lets not change it